store/sdk/src/main/java/org/apache/carbondata/sdk/file/arrow/ArrowConverter.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.ArrowFileWriter;
+import org.apache.arrow.vector.util.ByteArrayReadableSeekableByteChannel;
+
+public class ArrowConverter {
+
+  private final BufferAllocator allocator;
+  private VectorSchemaRoot root;
+  private ArrowWriter arrowWriter;
+  private org.apache.arrow.vector.types.pojo.Schema arrowSchema;
+  private ExtendedByteArrayOutputStream out;
+  private ArrowFileWriter writer;
+
+  public ArrowConverter(Schema schema, int initialSize) {
+    this.arrowSchema = ArrowUtils.toArrowSchema(schema, TimeZone.getDefault().getID());
+    this.allocator =
+        ArrowUtils.rootAllocator.newChildAllocator("toArrowBuffer", initialSize, Long.MAX_VALUE);
+    this.root = VectorSchemaRoot.create(arrowSchema, allocator);
+    this.arrowWriter = ArrowWriter.create(root);
+    // currently blocklet level read and set initial value to 32 MB.
+    this.out = new ExtendedByteArrayOutputStream(32 * 1024 * 1024);
+    this.writer = new ArrowFileWriter(root, null, Channels.newChannel(out));
+  }
+
+  /**
+   * write batch of row objects to Arrow vectors
+   *
+   * @param data
+   */
+  public void addToArrowBuffer(Object[] data) {
+    int i = 0;
+    while (i < data.length) {
+      arrowWriter.write((Object[]) data[i]);
+      i += 1;
+    }
+  }
+
+  /**
+   * To serialize arrow vectors to byte[]
+   *
+   * @return
+   * @throws IOException
+   */
+  public byte[] toSerializeArray() throws IOException {
+    arrowWriter.finish();
+    writer.writeBatch();
+    this.writer.close();
+    arrowWriter.reset();
+    writer.close();
+    this.root.close();
+    final byte[] bytes = out.toByteArray();
+    return bytes;
+  }
+
+  /**
+   * To copy arrow vectors to unsafe memory
+   *
+   * @return
+   * @throws IOException
+   */
+  public long copySerializeArrayToOffHeap() throws IOException {
+    arrowWriter.finish();
+    writer.writeBatch();
+    this.writer.close();
+    arrowWriter.reset();
+    writer.close();
+    this.root.close();
+    return out.copyToAddress();
+  }
+
+  /**
+   * Utility API to convert back the arrow byte[] to arrow VectorSchemaRoot.
+   *
+   * @param batchBytes
+   * @return
+   * @throws IOException
+   */
+  public VectorSchemaRoot byteArrayToVector(byte[] batchBytes) throws IOException {
+    ByteArrayReadableSeekableByteChannel in = new ByteArrayReadableSeekableByteChannel(batchBytes);
+    ArrowFileReader reader = new ArrowFileReader(in, allocator);
+    try {
+      VectorSchemaRoot root = reader.getVectorSchemaRoot();
+      VectorUnloader unloader = new VectorUnloader(root);
+      reader.loadNextBatch();
+      VectorSchemaRoot arrowRoot = VectorSchemaRoot.create(arrowSchema, allocator);
+      VectorLoader vectorLoader = new VectorLoader(arrowRoot);
+      vectorLoader.load(unloader.getRecordBatch());
+      return arrowRoot;
+    } catch (IOException e) {
+      reader.close();
+      throw new IOException(e);
```

Comment:
directly throw e no need to create new IOException object

Reply:
done